proxy/src/main/java/com/velocitypowered/proxy/config/VelocityConfiguration.java

```diff
@@ -503,6 +502,14 @@ public static VelocityConfiguration read(Path path) throws IOException {
     }
     forwardingSecret = forwardingSecretString.getBytes(StandardCharsets.UTF_8);
 
+    if (configVersion == 1.0 || configVersion == 2.0) {
+      config.set("force-key-authentication", config.getOrElse("force-key-authentication", true));
+      config.setComment("force-key-authentication",
+              "Should the proxy enforce the new public key security standard? By default, this is on.");
+      config.set("config-version", configVersion == 2.0 ? "2.5" : "1.5");
```

Comment:
There is no reason to move to a `.5` version without first having gone through `.1`, `.2`, `.3` and `.4` versions

Also, I think it is unnecessary to continue maintaining an old configuration version `1.x` since a single migration can be done by updating the old values and adding new configuration options to the configuration directly #788

Reply:
I honestly think that just generally, I wanna rewrite the way the config is handled as there's many oddities we've hit for god knows how long, many people seem to get burned by the fact that we're not writing the config out properly, etc https://github.com/parkervcp/eggs/issues/1741 - idk what exactly they're doing but there is defo some oddities induced here

Reply:
My egg should be pulling a perfectly valid config. I am not sure why it's doing what it's doing.

I will work on it soon. I have been moving my servers around the last few days.